src/celeritas/em/WentzelVIMscParams.cc

```diff
@@ -0,0 +1,80 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/em/WentzelVIMscParams.cc
+//---------------------------------------------------------------------------//
+#include "WentzelVIMscParams.hh"
+
+#include <algorithm>
+#include <utility>
+
+#include "corecel/io/Logger.hh"
+#include "corecel/sys/ScopedMem.hh"
+#include "celeritas/em/data/WentzelVIMscData.hh"
+#include "celeritas/io/ImportData.hh"
+#include "celeritas/mat/MaterialParams.hh"
+#include "celeritas/phys/ParticleParams.hh"
+
+#include "detail/MscParamsHelper.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct if Wentzel VI model is present, or else return nullptr.
+ */
+std::shared_ptr<WentzelVIMscParams>
+WentzelVIMscParams::from_import(ParticleParams const& particles,
+                                MaterialParams const& materials,
+                                ImportData const& data)
+{
+    auto is_wentzel = [](ImportMscModel const& imm) {
+        return imm.model_class == ImportModelClass::wentzel_vi_uni;
+    };
+    if (!std::any_of(data.msc_models.begin(), data.msc_models.end(), is_wentzel))
+    {
+        // No WentzelVI MSC present
+        return nullptr;
+    }
+    return std::make_shared<WentzelVIMscParams>(
+        particles, materials, data.msc_models);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from cross section data and material properties.
+ */
+WentzelVIMscParams::WentzelVIMscParams(ParticleParams const& particles,
+                                       MaterialParams const& materials,
+                                       VecImportMscModel const& mdata_vec)
+{
+    using units::MevEnergy;
+
+    ScopedMem record_mem("WentzelVIMscParams.construct");
+
+    HostVal<WentzelVIMscData> host_data;
+
+    MscParamsHelper helper(
+        particles, materials, mdata_vec, ImportModelClass::wentzel_vi_uni);
+    helper.build_ids(&host_data.ids);
+    helper.build_xs(&host_data.xs, &host_data.reals);
+
+    // Save electron mass
+    host_data.electron_mass = particles.get(host_data.ids.electron).mass();
+
+    // Save high/low energy limits
+    auto const& grid = host_data.xs[ItemId<XsGridData>(0)].log_energy;
+    host_data.params.low_energy_limit = MevEnergy(std::exp(grid.front));
+    host_data.params.high_energy_limit = MevEnergy(std::exp(grid.back));
```

Comment:
Maybe we should add emin/emax to the XsCalculator clsas? This is done here and in UrbanMsc and it's a bit low level ...